src/customPropTypes.js

```diff
@@ -10,25 +10,26 @@ const regexValidator = (regex, required) => (
   componentName,
 ) => {
   const value = props[propName]
-  if (!required && value === undefined) {
-    return null
+  const error = new Error(
+    `Invalid value: "${value}" of prop:"${propName}" supplied to ${componentName} component.`,
+  )
+
+  if (value === undefined) {
+    return !required ? null : error
```

Comment:
Always avoid using `negative if` when you have the `else`.
For instance, here:
```js
return required ? error : null

// clearer than:

return !required ? null : error
```

it is easier to understand and prettier ;)